elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreService.java

```diff
@@ -68,13 +74,20 @@ class KeyStoreService implements ModifiableKeyStoreService {
     private static final String GENERATED_CERTIFICATE_SIGNATURE_ALGORITHM = "SHA256withRSA";
     private static final int HEX_DELIMITER = ':';
     private static final String COMMON_NAME_PREFIX = "CN=";
+    //default delay between certificate health checks: 12h
+    static final long DEFAULT_DELAY = 12*60*60*1000;
 
+    private final String keyStoreName;
     private final String provider;
     private final String type;
     private final String path;
     private final String relativeTo;
     private final boolean required;
     private final String aliasFilter;
+    //delay in ms between service start and periodical check of certificate health
+    //if set to '0', this will mean one-time off warning, as prior to RFE
+    private long expirationCheckDelay = DEFAULT_DELAY;
+    private ScheduledExecutorService scheduledExecutorService;
```

Comment:
FYI as part of the CVE I am working on I am adding the scheduled executor service to the root of the Elytron subsystem so all Elytron services can share the same executor.